### PR TITLE
refactor: update integration tests workflow and add pytest configuration

### PR DESCRIPTION
**Pull request by magajh**
### Description

This PR includes some necessary changes to use the updated integration testing action https://github.com/eduNEXT/integration-test-in-tutor. The changes are described below:

- Added `pytest` and `pytest-django` requirements to run the tests.
- Updated the integration test workflow to use the updated action.
- Modified the target in the Makefile to use `pytest` for running unit tests, ensuring consistency in how tests are executed across all plugins.
- Added a target in the Makefile and a script to run the integration tests.
- Moved integration tests for the backends and the info view to the `integration` folders.

For more information about the changes and the reasoning behind them, please refer to the documentation on writing integration tests for the plugins: https://internal.docs.edunext.co/en/latest/internal/Teams/Descontrol/Internal_processes/integration_tests_in_plugins/integration_tests_with_openedx_components.html

### Note for Reviewers

This PR **does not move the unit test files** to `unit/` folders. That effort will be part of a future PR when integration tests are added to the plugin.

### How to Test

Ensure that both unit and integration tests pass without issues in the GitHub CI/CD pipeline.

### CODE REVIEW

**Review by BryanttV — APPROVED** [review]
Thanks @magajh, LGTM!

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -7,13 +7,12 @@ jobs:
...
-          path: eox-tenant
-      - uses: eduNEXT/integration-test-in-tutor@main
+      - name: Run Integration Tests
+        uses: eduNEXT/integration-test-in-tutor@mjh/make-extra-reqs-step-optional
         with:
           tutor_version: ${{ matrix.tutor_version }}
-          app_name: "eox-tenant"
-          shell_file_to_run: "eox_tenant/test/tutor/integration.sh"
+          app_name: 'eox-tenant'
+          shell_file_to_run: 'scripts/execute_integration_tests.sh'
```
Comment (BryanttV) [inline .github/workflows/integration-test.yml:17]: This input is necessary? since there is already a default with the same path in the action.

Comment (magajh) [reply .github/workflows/integration-test.yml:17]: @BryanttV you are right, I forgot we had a default for this, thanks for noticing!

**Review by MaferMazu — DISMISSED** [review]
My only comment here is we can use the main or release version of the integration test action. The rest looks good to me. Thanks ✨

**Comment by magajh** [comment]
@MaferMazu @BryanttV thank you both for the review! I've updated the action branch to main

**Review by BryanttV — APPROVED** [review]

**Review by MaferMazu — APPROVED** [review]

**Pull request merged**